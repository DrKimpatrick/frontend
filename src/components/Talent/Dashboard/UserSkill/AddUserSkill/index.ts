import AddUserSkill from './AddUserSkill';

export { AddUserSkill };
