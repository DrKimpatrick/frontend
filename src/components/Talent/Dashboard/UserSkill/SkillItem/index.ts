import SkillItem from './SkillItem';

export { SkillItem };
