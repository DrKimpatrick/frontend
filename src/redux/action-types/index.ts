// This is where we will add action types