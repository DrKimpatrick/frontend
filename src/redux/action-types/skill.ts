export enum SkillTypes {
  SkillLoading = 'Skill/Loading',
  SkillSuccess = 'Skill/SkillSuccess',
  SkillFail = 'Skill/SkillFail',
  GetSkillsLoading = 'Skill/Loading',
  GetSkillsSuccess = 'Skill/GetSkillsSuccess',
  GetSkillsFail = 'Skill/GetSkillsFail',
  ListUserSkill = 'Skill/ListUserSkill',
  Errors = 'Skill/Errors',
  Loading = 'Skill/LoadingSkill'
}

export enum VerificationStatus {
  Unverified = 'unverified',
  InProgress = 'inProgress',
  Verified = 'verified'
}

export interface Skill {
  _id: string;
  skill: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface UserSkill {
  _id: string;
  skill: Skill;
  createdAt?: string;
  updatedAt: string;
  level: string;
  verificationStatus: VerificationStatus;
}
export interface AddedSkill {
  _id: string;
  skill: string;
  user: string;
  level: string;
  verficationLevel: VerificationStatus;
}

export interface AddSkill {
  skill: string;
  level: string;
  verificationStatus: VerificationStatus;
}

export enum SkillLevel {
  Beginner = 'beginner',
  Advanced = 'advanced',
  Intermediate = 'intermediate'
}

export interface AddSkillLoading {
  type: SkillTypes.SkillLoading;
}

export interface AddSkillFail {
  type: SkillTypes.SkillFail;
  payload: string;
}

export interface AddSkillSuccess {
  type: SkillTypes.SkillSuccess;
  payload: AddedSkill;
}

export interface GetSkillsLoading {
  type: SkillTypes.GetSkillsLoading;
}

export interface GetSkillsFail {
  type: SkillTypes.GetSkillsFail;
  payload: string;
}

export interface GetSkillsSuccess {
  type: SkillTypes.GetSkillsSuccess;
  payload: Skill;
}

interface ListUserSkill {
  type: typeof SkillTypes.ListUserSkill;
  payload: {
    data: UserSkill[];
  };
}
interface Errors {
  type: typeof SkillTypes.Errors;
  payload: {
    errors: any;
  };
}

interface Loading {
  type: typeof SkillTypes.Loading;
  payload: {
    loading: boolean;
  };
}
export type SkillActionTypes =
  | AddSkillLoading
  | AddSkillSuccess
  | AddSkillFail
  | GetSkillsFail
  | GetSkillsSuccess
  | GetSkillsLoading
  | ListUserSkill
  | Errors
  | Loading;
