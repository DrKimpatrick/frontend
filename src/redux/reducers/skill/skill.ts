import {
  SkillTypes,
  SkillActionTypes,
  UserSkill,
  Skill
} from 'redux/action-types/skill';

interface InitialState {
  loading?: boolean;
  userSkill?: UserSkill[];
  errors?: any;
  skills?: Skill[];
  singleSkill?: Skill;
  action?: boolean;
  userSkillAction?: boolean;
  loadingBtn?: boolean;
}

const initialState: InitialState = {};

export const skillReducer = (
  state = initialState,
  action: SkillActionTypes
) => {
  switch (action.type) {
    case SkillTypes.ListUserSkill:
      return {
        ...state,
        userSkill: action.payload.data,
        loading: false,
        userSkillAction: false
      };

    case SkillTypes.Loading:
      return { ...state, loading: action.payload.loading };

    case SkillTypes.Errors:
      return { ...state, errors: action.payload.errors, loading: false };

    case SkillTypes.AddSkill:
      return {
        ...state,
        skills: state.skills
          ? state.skills.concat(action.payload.data)
          : action.payload.data,
        action: false
      };

    case SkillTypes.DeleteSkill:
      return {
        ...state,
        skills: state.skills
          ? state.skills.filter(item => item._id !== action.payload.data)
          : state.skills,
        action: false
      };

    case SkillTypes.UpdateSkill:
      if (state.skills) {
        const allSkills = state.skills;

        const skillIndex = allSkills.findIndex(
          item => item._id === action.payload.data._id
        );

        allSkills[skillIndex] = action.payload.data;

        return { ...state, skills: allSkills };
      }
      return state;

    case SkillTypes.ListSingleSkill:
      return { ...state, singleSkill: action.payload.data, loading: false };

    case SkillTypes.ListSkill:
      return {
        ...state,
        skills: action.payload.data,
        loading: false
      };

    case SkillTypes.SetActionModal:
      return { ...state, action: action.payload.data };

    case SkillTypes.AddUserSkill:
      return { ...state, userSkillAction: action.payload.data };

    case SkillTypes.LoadingBtn:
      return { ...state, loadingBtn: action.payload.loading };

    default:
      return state;
  }
};
