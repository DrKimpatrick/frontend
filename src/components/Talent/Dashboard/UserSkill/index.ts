import UserSkill from './UserSkill';

export { UserSkill };
export * from './SkillItem';
