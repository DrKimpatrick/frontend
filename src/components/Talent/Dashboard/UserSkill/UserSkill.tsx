import React, { useMemo, useEffect, useState } from 'react';
import { SkillLevel } from 'redux/action-types/skill';
import { FolderShared, Add } from '@material-ui/icons';
import { useSelector, useDispatch } from 'react-redux';
import { listUserSkill } from 'redux/actions/skill';
import { RootState } from 'redux/store';
import { SideLoading } from 'components/Reusable';
import { SkillItem } from '.';
import Headline from '../Headline';

const Skill = () => {
  const [loader, setLoader] = useState<boolean>();

  const dispatch = useDispatch();

  const reducer = useSelector((state: RootState) => {
    const { userSkill, loading } = state.skills;

    return { userSkill, loading };
  });

  const { userSkill } = reducer;

  useEffect(() => {
    setLoader(true);
    listUserSkill()(dispatch);
  }, [dispatch]);

  useEffect(() => {
    if (userSkill) {
      setLoader(false);
    }
  }, [userSkill]);

  const beginnerSkill = useMemo(() => {
    if (userSkill && userSkill.length > 0) {
      const listBeginnerSkill = userSkill.filter(
        item => item.level === SkillLevel.Beginner
      );

      if (listBeginnerSkill && listBeginnerSkill.length > 0) {
        return <SkillItem level="Beginner" userSkill={listBeginnerSkill} />;
      }
    }
    return <></>;
  }, [userSkill]);

  const intermediateSkill = useMemo(() => {
    if (userSkill && userSkill.length > 0) {
      const listIntermediateSkill = userSkill.filter(
        item => item.level === SkillLevel.Intermediate
      );

      if (listIntermediateSkill && listIntermediateSkill.length > 0) {
        return (
          <SkillItem level="Intermediate" userSkill={listIntermediateSkill} />
        );
      }
    }
    return <></>;
  }, [userSkill]);

  const advancedSkill = useMemo(() => {
    if (userSkill && userSkill.length > 0) {
      const listAdvancedSkill = userSkill.filter(
        item => item.level === SkillLevel.Advanced
      );

      if (listAdvancedSkill && listAdvancedSkill.length > 0) {
        return <SkillItem level="Advanced" userSkill={listAdvancedSkill} />;
      }
    }
    return <></>;
  }, [userSkill]);

  return (
    <div className="flex flex-col w-1/4 mt-12">
      <div>
        <Headline headline="SkillSet" icon={<FolderShared />} />
        {userSkill && userSkill.length > 0 && (
          <>
            {beginnerSkill}
            {intermediateSkill}
            {advancedSkill}
            <button
              className="mt-4 bg-gray-800 w-full text-white hover:bg-gray-900 outline-none font-semibold h-12 py-1 px-1 rounded-sm shadow"
              type="button"
            >
              <Add /> <span>Add new </span> <span>skills</span>
            </button>
          </>
        )}
        {userSkill && userSkill.length <= 0 && (
          <div className="notFound my-4">
            <h5>There are no skills</h5>
          </div>
        )}
        {loader && (
          <div className="my-10">
            <SideLoading size={30} />
          </div>
        )}
      </div>
    </div>
  );
};

export default Skill;
