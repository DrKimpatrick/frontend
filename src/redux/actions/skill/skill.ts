import { Dispatch } from 'redux';
import { SkillTypes } from 'redux/action-types/skill';
import { UserTypes } from 'redux/action-types/user';
import { setMessage, setError } from 'redux/actions/message';
import ApiAction from '../../../helpers/apiAction';

export const listUserSkill = () => (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: '/skills/me',
      method: 'GET',
      onStart: () => (dispatch: Dispatch) => {
        dispatch({ type: SkillTypes.Loading, payload: { loading: true } });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.ListUserSkill,
          payload: { data: res.data }
        });
        dispatch({ type: SkillTypes.Loading, payload: { loading: false } });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: { errors: error.response.data }
        });
        dispatch({ type: SkillTypes.Loading, payload: { loading: false } });
      }
    })
  );
};

export const changeSkillStatus = (values: { status: string; id: string }) => (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: `/skills/status/${values.id}`,
      method: 'PUT',
      data: { verificationStatus: values.status },
      onStart: () => dispatch => {
        dispatch({
          type: UserTypes.UserSkillLoading,
          payload: { loading: true }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.ChangeSkillStatus,
          payload: { data: res.data }
        });

        setMessage(res.message)(dispatch);
      }
    })
  );
};

export const setSkillAction = (value: boolean) => (dispatch: Dispatch) => {
  dispatch({ type: SkillTypes.SetActionModal, payload: { data: value } });
};

export const listSkill = () => async (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: '/skills',
      method: 'GET',
      onStart: () => dispatch => {
        dispatch({
          type: SkillTypes.Loading,
          payload: { loading: true }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.ListSkill,
          payload: { data: res.data }
        });
      }
    })
  );
};

export const listSingleSkill = (skillId: string) => async (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: `/skills/${skillId}`,
      method: 'GET',
      onStart: () => dispatch => {
        dispatch({
          type: UserTypes.UserSkillLoading,
          payload: { loading: true }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.ListSingleSkill,
          payload: { data: res.data }
        });
      }
    })
  );
};

export const deleteSkill = (skillId: string) => async (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: `/skills/${skillId}`,
      method: 'DELETE',
      onStart: () => dispatch => {
        dispatch({
          type: UserTypes.UserSkillLoading,
          payload: { loading: true }
        });
        setSkillAction(true)(dispatch);
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: () => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.DeleteSkill,
          payload: { data: skillId }
        });
        setSkillAction(false)(dispatch);
      }
    })
  );
};

export const updateSkill = ({
  skillId,
  skill
}: {
  skillId: string;
  skill: string;
}) => async (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: `/skills/${skillId}`,
      method: 'PUT',
      data: {
        skill
      },
      onStart: () => dispatch => {
        dispatch({
          type: UserTypes.UserSkillLoading,
          payload: { loading: true }
        });
        setSkillAction(true)(dispatch);
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.UpdateSkill,
          payload: { data: res.data }
        });

        setSkillAction(false)(dispatch);
      }
    })
  );
};

export const addSkill = (data: { skill: string }[]) => async (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: '/skills',
      method: 'POST',
      data,
      onStart: () => dispatch => {
        dispatch({
          type: UserTypes.UserSkillLoading,
          payload: { loading: true }
        });
        setSkillAction(true)(dispatch);
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.AddSkill,
          payload: { data: res.data }
        });

        setSkillAction(false)(dispatch);
      }
    })
  );
};

export const addUserSkill = (
  data: { skill: string; level: string }[]
) => async (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: '/skills/me',
      method: 'POST',
      data,
      onStart: () => dispatch => {
        dispatch({
          type: SkillTypes.LoadingBtn,
          payload: { loading: true }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
        dispatch({
          type: SkillTypes.LoadingBtn,
          payload: { loading: false }
        });
      },
      onSuccess: () => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.AddUserSkill,
          payload: { data: true }
        });

        dispatch({
          type: SkillTypes.LoadingBtn,
          payload: { loading: false }
        });
        listUserSkill()(dispatch);

        setMessage('skill saved successfully')(dispatch);
      }
    })
  );
};

export const updateUserSkill = (
  data: { userSkill: string; level: string }[]
) => async (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: '/skills/me',
      method: 'PATCH',
      data,
      onStart: () => dispatch => {
        dispatch({
          type: SkillTypes.LoadingBtn,
          payload: { loading: true }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
        dispatch({
          type: SkillTypes.LoadingBtn,
          payload: { loading: false }
        });
      },
      onSuccess: () => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.AddUserSkill,
          payload: { data: true }
        });

        dispatch({
          type: SkillTypes.LoadingBtn,
          payload: { loading: false }
        });
        listUserSkill()(dispatch);

        setMessage('skill updated successfully')(dispatch);
      }
    })
  );
};

export const deleteUserSkill = (value: string[]) => async (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: '/skills/me',
      method: 'DELETE',
      data: value,
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
        setError('Failed to delete skill')(dispatch);
      },
      onSuccess: () => (dispatch: Dispatch) => {
        listUserSkill()(dispatch);

        setMessage('deleted successfully')(dispatch);
      }
    })
  );
};
