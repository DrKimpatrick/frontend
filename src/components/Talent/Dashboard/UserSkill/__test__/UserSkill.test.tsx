import React from 'react';
import { Provider } from 'react-redux';
import { cleanup, render } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import ReactDom from 'react-dom';
import { Store } from 'redux';
import thunk from 'redux-thunk';
import { VerificationStatus, SkillLevel } from 'redux/action-types/skill';
import UserSkill from '../UserSkill';

let store: Store;

const middleware = [thunk];

const mockStore = configureMockStore(middleware);

const initialState = {
  skills: {
    userSkill: [
      {
        skill: {
          skill: 'php',
          createdAt: new Date().toDateString(),
          updatedAt: new Date().toDateString()
        },
        _id: 'id',
        level: SkillLevel.Beginner,
        verificationStatus: VerificationStatus.Unverified,
        createdAt: new Date().toDateString(),
        updatedAt: new Date().toDateString()
      },
      {
        skill: {
          skill: 'typescript',
          createdAt: new Date().toDateString(),
          updatedAt: new Date().toDateString()
        },
        _id: 'id',
        level: SkillLevel.Advanced,
        verificationStatus: VerificationStatus.Unverified,
        createdAt: new Date().toDateString(),
        updatedAt: new Date().toDateString()
      },
      {
        skill: {
          skill: 'typescript',
          createdAt: new Date().toDateString(),
          updatedAt: new Date().toDateString()
        },
        _id: 'id',
        level: SkillLevel.Intermediate,
        verificationStatus: VerificationStatus.Unverified,
        createdAt: new Date().toDateString(),
        updatedAt: new Date().toDateString()
      }
    ],
    loading: false
  }
};

describe('User Skill', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });

  afterEach(cleanup);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDom.render(
      <Provider store={store}>
        <UserSkill />
      </Provider>,
      div
    );
  });
  it('should create snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <UserSkill />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('Loading', () => {
    beforeEach(() => {
      store = mockStore({
        ...initialState,
        skills: { ...initialState.skills, loading: true }
      });
    });

    it('return SplashScreen component', () => {
      const { container } = render(
        <Provider store={store}>
          <UserSkill />
        </Provider>
      );

      const div: Element | any = container.querySelector(
        '.sideLoading'
      );

      expect(div).toBeDefined();
    });
  });

  describe('Not Found', () => {
    beforeEach(() => {
      store = mockStore({
        ...initialState,
        skills: { ...initialState.skills, userSkill: [] }
      });
    });

    it('return not found message when there is empty array', () => {
      const { container } = render(
        <Provider store={store}>
          <UserSkill />
        </Provider>
      );

      const h5: Element | any = container.querySelector('.notFound h5');

      expect(h5).toHaveTextContent('There are no skills');
    });
  });

  describe('Empty Fragment', () => {
    beforeEach(() => {
      store = mockStore({
        ...initialState,
        skills: { ...initialState.skills, userSkill: undefined }
      });
    });

    it('return empty when there is no user skills', () => {
      render(
        <Provider store={store}>
          <UserSkill />
        </Provider>
      );
    });
  });

});
