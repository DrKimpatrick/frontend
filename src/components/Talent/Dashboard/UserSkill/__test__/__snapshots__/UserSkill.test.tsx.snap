// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Skill should create snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full"
      >
        <p
          class="text-right text-xs py-2 pb-2 pr-3"
        >
          Beginner
        </p>
        <ul
          class="p-0 m-0 bg-card-preview "
        >
          <li
            class=" py-2 py-2 rounded-sm"
          >
            <div
              class="flex flex-row text-sm justify-between"
            >
              <h4
                class="medium-fonts px-4"
              >
                php
              </h4>
              <div
                class="flex flex-row"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root not-verified-color mr-4"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
                  />
                </svg>
              </div>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="w-full"
      >
        <p
          class="text-right text-xs py-2 pb-2 pr-3"
        >
          Intermediate
        </p>
        <ul
          class="p-0 m-0 bg-card-preview "
        >
          <li
            class=" py-2 py-2 rounded-sm"
          >
            <div
              class="flex flex-row text-sm justify-between"
            >
              <h4
                class="medium-fonts px-4"
              >
                php
              </h4>
              <div
                class="flex flex-row"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root not-verified-color mr-4"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
                  />
                </svg>
              </div>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="w-full"
      >
        <p
          class="text-right text-xs py-2 pb-2 pr-3"
        >
          Advanced
        </p>
        <ul
          class="p-0 m-0 bg-card-preview "
        >
          <li
            class=" py-2 py-2 rounded-sm"
          >
            <div
              class="flex flex-row text-sm justify-between"
            >
              <h4
                class="medium-fonts px-4"
              >
                php
              </h4>
              <div
                class="flex flex-row"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root not-verified-color mr-4"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
                  />
                </svg>
              </div>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="addButton"
      >
        <button
          class="mt-4 bg-gray-800 w-full text-white hover:bg-gray-900 outline-none font-semibold h-12 py-1 px-1 rounded-sm shadow add"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
           
          <span>
            Add new 
          </span>
           
          <span>
            skills
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full"
    >
      <p
        class="text-right text-xs py-2 pb-2 pr-3"
      >
        Beginner
      </p>
      <ul
        class="p-0 m-0 bg-card-preview "
      >
        <li
          class=" py-2 py-2 rounded-sm"
        >
          <div
            class="flex flex-row text-sm justify-between"
          >
            <h4
              class="medium-fonts px-4"
            >
              php
            </h4>
            <div
              class="flex flex-row"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root not-verified-color mr-4"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
                />
              </svg>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="w-full"
    >
      <p
        class="text-right text-xs py-2 pb-2 pr-3"
      >
        Intermediate
      </p>
      <ul
        class="p-0 m-0 bg-card-preview "
      >
        <li
          class=" py-2 py-2 rounded-sm"
        >
          <div
            class="flex flex-row text-sm justify-between"
          >
            <h4
              class="medium-fonts px-4"
            >
              php
            </h4>
            <div
              class="flex flex-row"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root not-verified-color mr-4"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
                />
              </svg>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="w-full"
    >
      <p
        class="text-right text-xs py-2 pb-2 pr-3"
      >
        Advanced
      </p>
      <ul
        class="p-0 m-0 bg-card-preview "
      >
        <li
          class=" py-2 py-2 rounded-sm"
        >
          <div
            class="flex flex-row text-sm justify-between"
          >
            <h4
              class="medium-fonts px-4"
            >
              php
            </h4>
            <div
              class="flex flex-row"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root not-verified-color mr-4"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
                />
              </svg>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="addButton"
    >
      <button
        class="mt-4 bg-gray-800 w-full text-white hover:bg-gray-900 outline-none font-semibold h-12 py-1 px-1 rounded-sm shadow add"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
         
        <span>
          Add new 
        </span>
         
        <span>
          skills
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
